Azure/sap-hana
# Sap lib + tfstate deployment

### Problem
Currently, tfstates are stored in local, which makes them vulnerable. 

### Solution
Seperate each module.
This pr focus on sap library: provision storage accounts and manage their tfstate.
/bootstrap/sap_library will be used to provision storage accounts for storing sapbits and tfstates
/run/sap_library will be used to config azurem as the backend of terraform and help to migrate the tfstates of storage accounts from local to remote storage account.

### Test
1. Logon to deployer vm, (checkout to this git branch) cd ~/sap-hana/deploy/terraform/bootstrap/sap_library
2. Edit sap_library.json.
3. Execute: terraform init
4. Execute: terraform apply -auto-approve -var-file=sap_library.json
5. After execution, there will be a shell script "tfstate_scripts.sh" generated.
6. Under "default" directory, execute: sh tfstate_scripts.sh 
7. Check if corresponding container has the saplibrary.terraform.tfstate file.
